tests/auxiliary/minicore.rs

```diff
@@ -0,0 +1,116 @@
+//! Auxiliary `minicore` prelude which stubs out std preludes for tests that need to work in
+//! cross-compilation scenarios where no `std` is available (that don't need to `-Zbuild-std`).
+
+#![feature(no_core, lang_items, rustc_attrs, unsized_fn_params, transparent_unions)]
+#![allow(unused, improper_ctypes_definitions, internal_features)]
+#![no_std]
+#![no_core]
+
+#[lang = "sized"]
+pub trait Sized {}
+
+#[lang = "receiver"]
+pub trait Receiver {}
+impl<T: ?Sized> Receiver for &T {}
+impl<T: ?Sized> Receiver for &mut T {}
+
+#[lang = "copy"]
+pub trait Copy: Sized {}
+impl Copy for i32 {}
+impl Copy for f32 {}
+impl<T: ?Sized> Copy for &T {}
+impl<T: ?Sized> Copy for *const T {}
+impl<T: ?Sized> Copy for *mut T {}
+
+#[lang = "clone"]
+pub trait Clone: Sized {
+    fn clone(&self) -> Self;
+}
+
+#[lang = "phantom_data"]
+pub struct PhantomData<T: ?Sized>;
+impl<T: ?Sized> Copy for PhantomData<T> {}
+
+#[lang = "unsafe_cell"]
+#[repr(transparent)]
+pub struct UnsafeCell<T: ?Sized> {
+    value: T,
+}
+
+pub trait Any: 'static {}
+
+pub enum Option<T> {
+    None,
+    Some(T),
+}
+impl<T: Copy> Copy for Option<T> {}
+
+pub enum Result<T, E> {
+    Ok(T),
+    Err(E),
+}
+impl<T: Copy, E: Copy> Copy for Result<T, E> {}
+
+#[lang = "manually_drop"]
+#[repr(transparent)]
+pub struct ManuallyDrop<T: ?Sized> {
+    value: T,
+}
+impl<T: Copy + ?Sized> Copy for ManuallyDrop<T> {}
+
+#[repr(transparent)]
+#[rustc_layout_scalar_valid_range_start(1)]
+#[rustc_nonnull_optimization_guaranteed]
+pub struct NonNull<T: ?Sized> {
+    pointer: *const T,
+}
+impl<T: ?Sized> Copy for NonNull<T> {}
+
+#[repr(transparent)]
+#[rustc_layout_scalar_valid_range_start(1)]
+#[rustc_nonnull_optimization_guaranteed]
+pub struct NonZero<T>(T);
```

Comment:
I think this will be way to easy to accidentally misuse without the `ZeroablePrimitive` trait bound. Maybe move it back into compatibility.rs until shown that other tests can also take advantage of it?

Reply:
Yeah, the initial version of minicore was basically copy-pasta from `tests/ui/abi/compatibility.rs`, I need to revisit and trim down the items it defines. (Minimal first, more items can be added later when converting other tests that want to use minicore on an as-needed basis.)

Reply:
Excluded the items with special attributes like `#[rustc_layout_scalar_valid_range_start(1)]` or `#[rustc_nonnull_optimization_guaranteed]` which are easy to abuse, and restored them to the original ABI test.